code/messaging/src/phone/java/com/adobe/marketing/mobile/MessageDelegate.java

```diff
@@ -0,0 +1,240 @@
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+ */
+
+package com.adobe.marketing.mobile;
+
+import static com.adobe.marketing.mobile.MessagingConstants.LOG_TAG;
+
+import android.webkit.ValueCallback;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+
+import com.adobe.marketing.mobile.services.ServiceProvider;
+import com.adobe.marketing.mobile.services.ui.AEPMessage;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessage;
+import com.adobe.marketing.mobile.services.ui.FullscreenMessageDelegate;
+import com.adobe.marketing.mobile.services.ui.MessageSettings;
+import com.adobe.marketing.mobile.services.ui.UIService;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class is the Messaging extension's internal implementation of the {@link FullscreenMessageDelegate}.
+ * Additionally, this class handles the dispatching of Experience Edge tracking events as well as opening URL's
+ * and loading any javascript code for the {@link Message} class.
+ */
+public class MessageDelegate implements FullscreenMessageDelegate {
+    private final static String SELF_TAG = "MessageDelegate";
+    private final static String AMPERSAND = "&";
+    private final static String EXPECTED_JAVASCRIPT_PARAM = "js=";
+    private final static String ADOBE_DEEPLINK = "adb_deeplink";
+    // public properties
+    public String messageId;
+    public boolean autoTrack = true;
+    // internal properties
+    MessagingInternal messagingInternal;
+    Map<String, Object> details;
+
+    /**
+     * Dispatch tracking information via a Messaging request content event.
+     *
+     * @param interactionType {@code String} containing the interaction which occurred
+     */
+    public void track(final String interactionType) {
+        if (StringUtils.isNullOrEmpty(interactionType)) {
+            Log.debug(LOG_TAG,
+                    "%s - Unable to record a message interaction - interaction string was null or empty.", SELF_TAG);
+            return;
+        }
+
+        final HashMap<String, Object> eventData = new HashMap<>();
+        eventData.put(MessagingConstants.EventDataKeys.Messaging.TRACK_INFO_KEY_EVENT_TYPE, MessagingConstants.EventDataKeys.Messaging.IAMDetailsDataKeys.EventType.INTERACT);
+        eventData.put(MessagingConstants.EventDataKeys.Messaging.TRACK_INFO_KEY_MESSAGE_EXECUTION_ID, messageId);
+        eventData.put(MessagingConstants.EventDataKeys.Messaging.TRACK_INFO_KEY_ACTION_ID, interactionType);
+
+        final Event event = new Event.Builder(MessagingConstants.EventName.MESSAGING_IAM_TRACKING_MESSAGING_EVENT, MessagingConstants.EventType.MESSAGING, MessagingConstants.EventSource.REQUEST_CONTENT).setEventData(eventData).build();
```

Comment:
Can we shorten the constant name `MESSAGING_IAM_TRACKING_MESSAGING_EVENT`.
imo `EventName.IAM_TRACKIN` should suffice.